src/bootstrap/lib.rs

```diff
@@ -517,10 +521,13 @@ impl Build {
     /// The mode indicates what the root directory is for.
     fn stage_out(&self, compiler: Compiler, mode: Mode) -> PathBuf {
         let suffix = match mode {
-            Mode::Libstd => "-std",
-            Mode::Libtest => "-test",
-            Mode::Tool => "-tools",
-            Mode::Librustc => "-rustc",
+            Mode::Std => "-std",
+            Mode::ToolStd => "-tools",
+            Mode::Test => "-test",
+            Mode::ToolTest => "-tools",
+            Mode::Codegen => "-rustc",
+            Mode::Rustc => "-rustc",
+            Mode::ToolRustc => "-tools",
```

Comment:
Let's instead combine all of the tools into one pattern or have them after each other so it's clearer what's going on